trace/google/cloud/trace/span_context.py

```diff
@@ -0,0 +1,164 @@
+# Copyright 2017 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""SpanContext encapsulates the current context within the request's trace."""
+
+from google.cloud.trace.trace import generate_trace_id
+
+import re
+import logging
+
+_TRACE_CONTEXT_HEADER_FORMAT = '([0-9a-f]{32})(\/(\d+))?(;o=(\d+))?'
+_TRACE_HEADER_KEY = 'X_CLOUD_TRACE_CONTEXT'
+_TRACE_ID_FORMAT = '[0-9a-f]{32}?'
+
+
+class SpanContext(object):
+    """SpanContext includes 3 fields: traceId, spanId, and an enabled flag 
+    which indicates whether or not the request is being traced. It contains the
+    current context to be propagate to the child spans.
+    
+    :type trace_id: str
+    :param trace_id: (Optional) Trace_id is a 32 digits uuid for the trace.
+                     If not given, will generate one automatically.
+
+    :type span_id: int
```

Comment:
To avoid endianness problems I would avoid using ints. Probably better to do something similar with https://github.com/census-instrumentation/opencensus-java/blob/master/api/src/main/java/io/opencensus/trace/SpanId.java